src/core/hle/kernel/memory/page_heap.h

```diff
@@ -0,0 +1,386 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+// This file references various implementation details from Atmosph�re, an open-source firmware for
+// the Nintendo Switch. Copyright 2018-2020 Atmosph�re-NX.
+
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "common/alignment.h"
+#include "common/assert.h"
+#include "common/common_funcs.h"
+#include "common/common_types.h"
+#include "core/hle/kernel/memory/memory_types.h"
+
+namespace Kernel::Memory {
+
+class PageHeap final : NonCopyable {
+private:
+    static constexpr inline std::size_t MemoryBlockPageShifts[]{0xC,  0x10, 0x15, 0x16,
+                                                                0x19, 0x1D, 0x1E};
+    static constexpr std::size_t NumMemoryBlockPageShifts{7};
+
+public:
+    static constexpr s32 GetAlignedBlockIndex(std::size_t num_pages, std::size_t align_pages) {
+        const std::size_t target_pages{std::max(num_pages, align_pages)};
+        for (std::size_t i{}; i < NumMemoryBlockPageShifts; i++) {
```

Comment:
```suggestion
        for (std::size_t i = 0; i < NumMemoryBlockPageShifts; i++) {
```
Here is much more idiomatic and explicit in what it means compared to just bracing. Especially as a loop variable.